### DISCUSSION TITLE
Method parameter types using CLI

### DISCUSSION THREAD

**Discussion by andreinaku** [microsoft/pyright#8938] [Q&A] [answered]
Hello!
Is there a parameter for the CLI to add class method inferred parameter types? I am looking at the following example:
```
class foo:
    def __init__(self, x):
        self.x = x
    
    def bar(self):
        self.y = 10


f = foo(10)
```
and the result after running `pyright --createstub test` (test is the name of the file) is:
```
"""
This type stub file was generated by pyright.
"""

class foo:
    def __init__(self, x) -> None:
        ...
    
    def bar(self): # -> None:
        ...
    

f = ...

```
I would have expected `x: Any` as an annotation. Also, `f: foo` would have been another.

PS: is there also any way for the CLI to output inferred types within method/function bodies?

**Comment by erictraut** [comment, marked as answer]
No, there's no option like that. Adding `Any` parameter annotations is not recommended in this case. It obscures the fact that the parameter is missing type information and leads to a false sense of type safety for users of the stub. It is recommended that stubs that are missing type information simply omit the corresponding annotation. When such a stub is used in pyright, these missing annotations will be translated to an `Unknown` type rather than `Any`, and this will tell users that they are at risk because of missing type information.

> is there also any way for the CLI to output inferred types within method/function bodies

Symbols defined within a function body (local variables) are never emitted within a stub. Symbols defined within a class body are emitted, but their inferred type is not generated by pyright. Inferred types can be inaccurate, so it's best to add those by hand so they match the real intent of the library author.
